File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.java
class Solution {
    public int subarraySum(int[] nums, int k) {
        int count=0;
        int sum=0;
        Map<Integer,Integer> prefixSum = new HashMap<>();
        prefixSum.put(0,1);
        for(int i=0;i<nums.length;i++){
            sum += nums[i];
            if(prefixSum.containsKey(sum-k)){
                count += prefixSum.get(sum-k);
            }
            prefixSum.put(sum,prefixSum.getOrDefault(sum,0)+1);
        }
        return count;
    }
}

File: 0560-subarray-sum-equals-k/0560-subarray-sum-equals-k.rs
use std::collections::HashMap;

impl Solution {
    pub fn subarray_sum(nums: Vec<i32>, k: i32) -> i32 {
        let mut freq: HashMap<i64, i32> = HashMap::new(); 
        freq.insert(0,1);
        let mut sum: i64 = 0;
        let mut count: i32=0;
        for n in nums {
            sum += n as i64;
            let need = sum - k as i64;
            if let Some(&c) = freq.get(&need){
                count += c;
            }
            *freq.entry(sum).or_insert(0) += 1;
        }
        count
    }
}